src/ParseServer.js

```diff
@@ -412,16 +416,26 @@ class ParseServer {
     return new ParseLiveQueryServer(httpServer, config);
   }
 
-  static verifyServerUrl(serverUrl) {
-    // perform a health check on the provided serverURL
-    if(serverUrl) {
-      const request = require('request');
-      request(serverUrl.replace(/\/$/, "") + "/health", function (error, response, body) {
-        if (error || response.statusCode !== 200 || body !== "OK") {
-          console.error("\nWARNING, Unable to connect to publicServerURL '" + serverUrl + "'. Cloud code and push notifications may be unavailable!\n");
-        }
-      });
-    }
+  static verifyServerUrl(serverUrl, callback) {
+    // perform a health check on the publicServerURL value, with 2.5 second delay
+    setTimeout(function() {
+      if(serverUrl) {
+        const request = require('request');
+        request(serverUrl.replace(/\/$/, "") + "/health", function (error, response, body) {
```

Comment:
you could use Parse._request probably this way you don't even need those replacements.

Reply:
And I'll look to see if that'll swap out ok.